diff --git a/BUILD.gn b/BUILD.gn
index 40ac574..6f5e4f7 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -555,6 +555,12 @@ if (!build_with_chromium) {
       "pc:libjingle_peerconnection",
       "pc:rtc_pc",
       "video",
+      "//third_party/zlib",
+      "rtc_base:log_sinks",
+      "rtc_base:socket_adapters",
+      "media:rtc_simulcast_encoder_adapter",
+      "api:enable_media_with_defaults",
+      "api:field_trials",
     ]
 
     if (rtc_include_builtin_audio_codecs) {
