diff --git a/video/video_send_stream_impl.cc b/video/video_send_stream_impl.cc
index 1120e92..5e39064 100644
--- a/video/video_send_stream_impl.cc
+++ b/video/video_send_stream_impl.cc
@@ -583,6 +583,7 @@ VideoSendStreamImpl::~VideoSendStreamImpl() {
   RTC_LOG(LS_INFO) << "~VideoSendStreamImpl: " << config_.ToString();
   RTC_DCHECK(!started());
   RTC_DCHECK(!IsRunning());
+  video_stream_encoder_->Stop();
   transport_->DestroyRtpVideoSender(rtp_video_sender_);
 }
 
