diff --git a/media/engine/webrtc_video_engine.cc b/media/engine/webrtc_video_engine.cc
index 256a5aa..46d96b3 100644
--- a/media/engine/webrtc_video_engine.cc
+++ b/media/engine/webrtc_video_engine.cc
@@ -3295,8 +3295,11 @@ void WebRtcVideoReceiveChannel::OnPacketReceived(RtpPacketReceived packet) {
 
   call_->Receiver()->DeliverRtpPacket(
       MediaType::VIDEO, std::move(packet),
-      absl::bind_front(
-          &WebRtcVideoReceiveChannel::MaybeCreateDefaultReceiveStream, this));
+      [this, safety = task_safety_.flag()](
+          const RtpPacketReceived& undemuxable_packet) {
+        return safety->alive() &&
+               MaybeCreateDefaultReceiveStream(undemuxable_packet);
+      });
 }
 
 bool WebRtcVideoReceiveChannel::MaybeCreateDefaultReceiveStream(
diff --git a/media/engine/webrtc_voice_engine.cc b/media/engine/webrtc_voice_engine.cc
index 312c6f0..0e8c040 100644
--- a/media/engine/webrtc_voice_engine.cc
+++ b/media/engine/webrtc_voice_engine.cc
@@ -2605,8 +2605,11 @@ void WebRtcVoiceReceiveChannel::OnPacketReceived(RtpPacketReceived packet) {
 
   call_->Receiver()->DeliverRtpPacket(
       MediaType::AUDIO, std::move(packet),
-      absl::bind_front(
-          &WebRtcVoiceReceiveChannel::MaybeCreateDefaultReceiveStream, this));
+      [this, safety = task_safety_.flag()](
+          const RtpPacketReceived& undemuxable_packet) {
+        return safety->alive() &&
+               MaybeCreateDefaultReceiveStream(undemuxable_packet);
+      });
 }
 
 bool WebRtcVoiceReceiveChannel::MaybeCreateDefaultReceiveStream(
